### PR TITLE
FP16MomentumSGDUpdate Op fix and enable for ROCm

### PR DESCRIPTION
**Pull request by rohithkrn**
1. Fix a bug in FP16MomentumSGDUpdate operator
2. Enable operator for ROCm

### CODE REVIEW

**Comment by rohithkrn** [comment]
@bddppq @petrex

File: caffe2/python/operator_test/momentum_sgd_test.py [hunk 1]
```diff
@@ -174,11 +174,13 @@ def momentum_sgd(grad, param_momentum, lr, param=None):
             weight_decay=0.0,
         )
 
+        threshold = 1e-3 if (gc.device_type == caffe2_pb2.HIP) else 1e-4
```
Comment (petrex) [inline caffe2/python/operator_test/momentum_sgd_test.py]: @bddppq Is it a valid threshold? pls advise.

Comment (bddppq) [reply caffe2/python/operator_test/momentum_sgd_test.py]: The (random) inputs could be different on different machines. When the tests failed, "hypothesis" library should print the "seed" for reproduce the same failure, could you paste that here?

Comment (bddppq) [reply caffe2/python/operator_test/momentum_sgd_test.py]: @petrex I'm hoping you can tell me whether the threshold is valid :-)

**Review by bddppq — APPROVED** [review]
good catch

**Pull request closed without merging**